### PR TITLE
OT-2044 Fix problem with multiple blenders on windows via pid

### PR DESCRIPTION
**Pull request by amilovsky**

### CODE REVIEW

File: keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py [hunk 1]
```diff
@@ -314,18 +319,59 @@ def is_installed(force_recheck=False):
...
+    return pids
+
+
+def _all_dirs_in_dir(dir):
+    return [f for f in os.listdir(dir) if os.path.isdir(os.path.join(dir, f))]
+
+
+def _remove_old_dirs(base_dir):
+    pids = _get_all_pids_on_windows()
+    dirs = _all_dirs_in_dir(base_dir)
```
Comment (SergeyKrivohatskiy) [inline keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py:352]: swap pids= and dirs=  lines here. 
```
pids = []
# new process `1` starts and creates dir
dir = ['1'] # this dir gets removed
```
False positive is a bug here, false negative is OK

File: keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py [hunk 2]
```diff
@@ -314,18 +319,59 @@ def is_installed(force_recheck=False):
...
+        shutil.rmtree(dir, ignore_errors=True)
+    except Exception as err:
+        logger = logging.getLogger(__name__)
+        logger.error('remove_dir: Cannot delete dir {}'.format(dir))
+        logger.error('Exception info: {}'.format(str(err)))
+
+
+def _get_all_pids_on_windows():
+    output = subprocess.run(['tasklist', '/NH', '/FO', 'CSV'],
+                            capture_output=True)  # windows specific code
```
Comment (SergeyKrivohatskiy) [inline keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py:333]: Remove comment. Maybe make assert if you need to make a point here

File: keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py [hunk 3]
```diff
@@ -314,18 +319,59 @@ def is_installed(force_recheck=False):
...
+    try:
+        shutil.rmtree(dir, ignore_errors=True)
+    except Exception as err:
+        logger = logging.getLogger(__name__)
+        logger.error('remove_dir: Cannot delete dir {}'.format(dir))
+        logger.error('Exception info: {}'.format(str(err)))
+
+
+def _get_all_pids_on_windows():
+    output = subprocess.run(['tasklist', '/NH', '/FO', 'CSV'],
```
Comment (SergeyKrivohatskiy) [inline keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py:332]: Any checks here?

**Review by SergeyKrivohatskiy — CHANGES_REQUESTED** [review]

**Review by SergeyKrivohatskiy — APPROVED** [review]

File: keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py (hunk 1 above)
Comment (amilovsky) [reply keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py:361]: I agree. Fixed

File: keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py (hunk 2 above)
Comment (amilovsky) [reply keentools_facebuilder/blender_independent_packages/pykeentools_loader/install.py]: Wrapped in try-except

**Pull request merged**